sql/parser/parse_test.go

```diff
@@ -275,9 +275,14 @@ func TestParse2(t *testing.T) {
 			`SELECT CAST('1' AS INT)`},
 		// Double negation. See #1800.
 		{`SELECT *,-/* comment */-5`,
-			`SELECT *, - -5`},
+			`SELECT *, - - 5`},
 		{"SELECT -\n-5",
-			`SELECT - -5`},
+			`SELECT - - 5`},
+		{"SELECT 1e-\n-1",
```

Comment:
I think this is a bug in `scanner.scanNumber()`. The postgres scanner has the following rules for scanning floating point numbers:

```
real			({integer}|{decimal})[Ee][-+]?{digit}+
realfail1		({integer}|{decimal})[Ee]
realfail2		({integer}|{decimal})[Ee][-+]
```

If the scanner sees `-` or `+` it should require the following character to be a digit.